src/main/java/org/gridsuite/study/server/service/StudyService.java

```diff
@@ -1353,6 +1354,40 @@ public void moveStudyNode(UUID studyUuid, UUID nodeToMoveUuid, UUID referenceNod
         notificationService.emitElementUpdated(studyUuid, userId);
     }
 
+    @Transactional
+    public void duplicateStudySubtree(UUID studyUuid, UUID parentNodeToCopyUuid, UUID referenceNodeUuid, String userId) {
+        checkStudyContainsNode(studyUuid, parentNodeToCopyUuid);
+        checkStudyContainsNode(studyUuid, referenceNodeUuid);
+
+        UUID duplicatedNodeUuid = networkModificationTreeService.duplicateStudySubtree(parentNodeToCopyUuid, referenceNodeUuid, new HashSet<>());
+        notificationService.emitSubtreeInserted(studyUuid, duplicatedNodeUuid, referenceNodeUuid);
+        notificationService.emitElementUpdated(studyUuid, userId);
+    }
+
+    @Transactional
+    public void moveStudySubtree(UUID studyUuid, UUID parentNodeToMoveUuid, UUID referenceNodeUuid, String userId) {
+        checkStudyContainsNode(studyUuid, parentNodeToMoveUuid);
+        checkStudyContainsNode(studyUuid, referenceNodeUuid);
+
+        List<UUID> allChildren = networkModificationTreeService.getChildren(parentNodeToMoveUuid);
+        if (allChildren.contains(referenceNodeUuid)) {
+            throw new StudyException(NOT_ALLOWED);
+        }
+        networkModificationTreeService.moveStudySubtree(parentNodeToMoveUuid, referenceNodeUuid);
+
+        if (networkModificationTreeService.getBuildStatus(parentNodeToMoveUuid) == BuildStatus.BUILT) {
+            updateStatuses(studyUuid, parentNodeToMoveUuid, false, true);
+        }
+        allChildren.forEach(childUuid -> {
+            if (networkModificationTreeService.getBuildStatus(childUuid) == BuildStatus.BUILT) {
+                updateStatuses(studyUuid, childUuid, false, true);
+            }
+        });
```

Comment:
Maybe we can do it like that:
allChildren.stream()
                .filter(childUuid ->  networkModificationTreeService.getBuildStatus(childUuid) == BuildStatus.BUILT)
                .forEach(childUuid -> updateStatuses(studyUuid, childUuid, false, true));